azure-sdk-build-tool/src/main/java/com/microsoft/azure/sdk/build/tool/mojo/AzureSdkMojo.java

```diff
@@ -0,0 +1,253 @@
+import com.microsoft.azure.sdk.build.tool.models.BuildErrorLevel;
+import com.microsoft.azure.sdk.build.tool.models.BuildReport;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Azure SDK build tools Maven plugin Mojo for analyzing Maven configuration of an application to provide Azure
+ * SDK-specific recommendations.
+ */
+@Mojo(name = "run",
+    defaultPhase = LifecyclePhase.PREPARE_PACKAGE,
+    requiresDependencyCollection = ResolutionScope.RUNTIME,
+    requiresDependencyResolution = ResolutionScope.RUNTIME)
+public class AzureSdkMojo extends AbstractMojo {
+
+    public static final TypeReference<Map<String, Object>> MAP_TYPE_REFERENCE = new TypeReference<Map<String, Object>>() {
+    };
+    private static AzureSdkMojo mojo;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final String APP_INSIGHTS_INSTRUMENTATION_KEY = "1d377c0e-44f8-4d56-bee7-7f13a3fef594";
+    private static final String APP_INSIGHTS_ENDPOINT = "https://centralus-2.in.applicationinsights.azure.com/";
+    private static final String AZURE_SDK_BUILD_TOOL = "azure-sdk-build-tool";
+    private final ApplicationInsightsClient applicationInsightsClient;
+
+    @Parameter(defaultValue = "${project}", readonly = true, required = true)
+    private MavenProject project;
+
+    @Parameter(property = "validateAzureSdkBomUsed", defaultValue = "true")
+    private boolean validateAzureSdkBomUsed;
+
+    @Parameter(property = "validateNoDeprecatedMicrosoftLibraryUsed", defaultValue = "true")
+    private boolean validateNoDeprecatedMicrosoftLibraryUsed;
+
+    @Parameter(property = "validateBomVersionsAreUsed", defaultValue = "true")
+    private boolean validateBomVersionsAreUsed;
+
+    @Parameter(property = "validateNoBetaLibraryUsed", defaultValue = "true")
+    private boolean validateNoBetaLibraryUsed;
+
+    @Parameter(property = "validateNoBetaApiUsed", defaultValue = "true")
+    private boolean validateNoBetaApiUsed;
+
+    @Parameter(property = "validateLatestBomVersionUsed", defaultValue = "true")
+    private boolean validateLatestBomVersionUsed;
+
+    @Parameter(property = "reportFile", defaultValue = "")
+    private String reportFile;
+
+    @Parameter(property = "sendToMicrosoft", defaultValue = "true")
+    private boolean sendToMicrosoft;
```

Comment:
Would it be more compliant to ask the user if they would like to send data to Microsoft on the first run?